### PR TITLE
Fix refresrate changing

### PR DESCRIPTION
**Pull request by fritsch**
During below additions the logic was flawed in a sense that 3:2 pulldown was evaluated too early. Means if you run at 60hz it would never switch to 24 hz and if you run at 59.94 hz it would never switch to 23.976, etc.

This reverts the mixed in logic, corrects the prints and uses the 3:2 as a final path.

### CODE REVIEW

**Comment by fritsch** [comment]
Forum reports:
https://forum.kodi.tv/showthread.php?tid=343524
https://forum.kodi.tv/showthread.php?tid=343518

File: xbmc/windowing/Resolution.cpp [hunk 1]
```diff
@@ -105,7 +103,7 @@ void CResolutionUtils::FindResolutionFromWhitelist(float fps, int width, int hei
 
   CLog::Log(LOGDEBUG, "Trying to find exact refresh rate");
 
-  for (const auto& mode : indexList)
+  for (const auto &mode : indexList)
```
Comment (lrusak) [inline xbmc/windowing/Resolution.cpp]: why did you change this? we left align in Kodi :stuck_out_tongue_closed_eyes:

Comment (fritsch) [reply xbmc/windowing/Resolution.cpp]: In order to properly revert it. I will add it as a cosmetics commit

**Comment by fritsch** [comment]
@lrusak okay with the added commit?

**Comment by lrusak** [comment]
> @lrusak okay with the added commit?

Yes thanks :) I didn't realize that the the code was part of a revert initially

Also, my 2 cents, I feel that the whitelist has become abused and turned into something it wasn't supposed to be (the catch all solution like we had before). The whole point of the whitelist was to add control and predictability to mode switching. I feel that we have lost this.

I think we need to look at it again and take a look at all the possible problems with mode switching and what solutions are available. If we need to add an extra setting or some other custom controls then so be it. We cannot keep adding exceptions to the whitelist.

**Comment by fritsch** [comment]
I concurr here a bit.

Whitelist: Add modes you like to switch to
Resolution.cpp: Find the best mode depending on the selected modes. This is currently: direct, double, pulldown.

What exception do you see?

**Comment by fritsch** [comment]
After discussion with @peak3d I close this one. There is no sense in deciding between "half bullshittish solutions" and "40% bullshittish solutions".

For Leia - please decide what the team likes best :-)
